edit/PersistentVolumeClaim.vue

```diff
@@ -0,0 +1,104 @@
+<script>
+import LabeledSelect from '@/components/form/LabeledSelect';
+import UnitInput from '@/components/form/UnitInput';
+import RadioGroup from '@/components/form/RadioGroup';
+import Checkbox from '@/components/form/Checkbox';
+import LabeledInput from '@/components/form/LabeledInput';
+import { mapGetters } from 'vuex';
+import { removeObject } from '@/utils/array';
+import { STORAGE_CLASS, PV } from '@/config/types';
+import { allHash } from '@/utils/promise';
+export default {
+
+  components: {
+    LabeledSelect, UnitInput, RadioGroup, Checkbox, LabeledInput
+  },
+  props:      {
+    mode: {
+      type:    String,
+      default: 'create'
+    },
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+    registerBeforeHook: {
+      type:    Function,
+      default: null,
+    },
+  },
+  async fetch() {
+    const hash = await allHash({
+      storageClasses:    this.$store.dispatch('cluster/findAll', { type: STORAGE_CLASS }),
+      persistentVolumes: this.$store.dispatch('cluster/findAll', { type: PV }),
+    });
+
+    this.storageClasses = hash.storageClasses.map(sc => sc.metadata.name);
+    this.persistentVolumes = hash.persistentVolumes.map(pv => pv.metadata.name);
+  },
+
+  data() {
+    return {
+      storageClasses: [], persistentVolumes: [], createPVC: true
+    };
+  },
+
+  computed: { ...mapGetters({ t: 'i18n/t' }) },
+
+  created() {
+    if (!this.value.spec) {
+      this.$set(this.value, 'spec', { accessModes: [], resources: { requests: { storage: null } } });
+    }
```

Comment:
This is ok here but you can also do it in the model's applyDefaults so that they're always setup when a PVC model is created.  (example in https://github.com/rancher/dashboard/blob/60f44836fd903815eb0fcdbe7f34b98b26e4e6b6/models/catalog.cattle.io.release.js#L13)